bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -0,0 +1,277 @@
+    col_names: The list of columns in rdf 
+    func: The python callable that is to be jitted.
+    params: List of parameters of the above callable
+    func_args: Dict to store the value of each param
+    args_info: Information about the type and value of each param
+    func_sign: Contains the function signature
+    func_call: Contains the function call as a string.
+
+    Methods
+    -------
+    find_type(x): determines the type of the variable passed
+    find_function_params(func): determines the parameter list of the python callable passed
+    generate_func_args(col_list, extra_args): uses the user passed column_list and extra_args to determine the value and type of each parameter of the callable. Stores it in arg_info
+    find_function_signature(): Calculates arg_info containing type  and value of each parameter and stores it in arg_info
+    generate_function_call(): Uses the arg_info to create a NumbaDeclare compatible function signature and the corresponding function call.
+    jit_function(func, cols_list, extra_args): Uses the above functions to jit the function and return's its call   
+
+    eg:
+
+    """
+
+    function_cache = {} # Variable to store previous functions so as to not rejit.
+    lambda_function_counter = 0 # Counter to name the lambda functions
+    
+    def __init__(self, rdf) -> None:
+        self.rdf = rdf
+        self.col_names = rdf.GetColumnNames()
+    
+    def find_type(self, x):
+        """
+        Function to determine the type of a variable x.
+        If x is a string. It maps it to the corresponding column's type in RDF. (String constants are not supported)
+        Else it determines the fundamental type of x. (int, float, bool)
+        Else it is an a numpy array and maps it to corresponding RVec.
+        Otherwise flags a type error
+        Args:
+	    `   x: Variable whose type is to be determined
+
+        Returns:
+            type of the variable x
+        """
+        try:
+            import numpy as np
+            from ._rdf_conversion_maps import FUNDAMENTAL_PYTHON_TYPES, TREE_TO_NUMBA, NUMPY_TO_TREE
+        except:
+            raise ImportError("Failed to import numpy during call to determine function signature.")
+        if isinstance(x, str):
+            #Can be string constant or can be column name
+            if x in self.col_names: # If x is a column
+                t = self.rdf.GetColumnType(x)
+                if t in TREE_TO_NUMBA: # The column is a fundamental type from tree  
+                    return TREE_TO_NUMBA[t]
+                elif '<' in  t: # The column type is a RVec<type>
+                    g = re.match('(.*)<(.*)>', t).groups(0)
+                    if g[1] in TREE_TO_NUMBA:
+                        return "RVec<" + TREE_TO_NUMBA[g[1]] + ">"
+                    return "RVec<" + str(g[1]) + ">" # There are data type that leak into here. Not sure from where. But need to implement something here such that this condition is never met.
```

Comment:
What about columns with type `RVec<RVec<RVec<float>>>` ? Since we don't support them for Numba declare, we should also error out here. You can check how many `RVec`s are in the column type and if there's more than one, you can error out like `TypeError(f"Only columns with 'RVec<T>' of a fundamental type are supported, not '{t}'")`